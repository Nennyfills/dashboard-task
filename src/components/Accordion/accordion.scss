.accordion-row {
 &__col {
  input {
   position: absolute;
   opacity: 0;
   z-index: -1;
  }
 }
 &--tab {
  width: 100%;
  @include respond(phone) {
   width: auto;
  }
  color: #fff;
 }
 .content {
  height: 1em;
  min-height: 0;
  padding: 0 1em;
  transition: all 0.35s;
  transform: scale(0.01);
  opacity: 0;
 }

 label {
  @include display-flex();
  padding: 20px;
  background: $white;
  text-transform: capitalize;
  cursor: pointer;
  border-radius: 0.4em;
  background: $white;
  border-radius: 10px;
 }
 input:checked {
  color: pink;
  + .Label {
   background: darken($primary, 10%);
   color: $white;

   &::after {
    transform: rotate(90deg);
   }
  }
  ~ .content {
   height: fit-content;
   overflow: auto;
   margin-bottom: 1em;
   padding: 1em;
   transform: scale(1);
   opacity: 1;
  }
 }
}
