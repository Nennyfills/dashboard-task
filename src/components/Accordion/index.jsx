import Loader from 'components/Loader';
import currencyFormat from 'helpers';
import PropTypes from 'prop-types';
import { useEffect } from 'react';

const Accordion = ({ children, titleData, loading }) => {
  useEffect(() => {
    const input = document.getElementsByClassName('accordion')[0];
    if (input) input.checked = true;
  }, []);

  return (
    <div className="accordion-row" data-testid="accordion-row">
      <div className="accordion-row__col" data-testid="accordion-row__col">
        <div className="accordion-row__tabs" data-testid="accordion-row__tabs">
          {children?.map((val, index) => (
            <div
              key={`${titleData[index]?.name}-${index}`}
              className="accordion-row__tab"
              data-testid="accordion-row__tabs"
            >
              <input type="checkbox" className="accordion" id={`check-${index}`} />
              {loading ? <Loader notTable /> : (
                <label
                  className="accordion-row__tab--label semibold"
                  htmlFor={`check-${index}`}
                  date-label="amount"
                  data-testid="accordion-row__tab--label"
                >
                  <p className="bold">{titleData[index]?.name}</p>
                  {titleData[index]?.total && (
                  <p className="bold">
                    <span>Total: </span>
                    {currencyFormat(titleData[index]?.total)}
                  </p>
                  )}
                </label>
              )}
              <div className="content">
                <div className="content--inner">{val}</div>
              </div>
            </div>
          ))}
        </div>
      </div>
    </div>
  );
};

Accordion.propTypes = {
  children: PropTypes.oneOfType([PropTypes.string, PropTypes.node, PropTypes.bool]),
  titleData: PropTypes.oneOfType([
    PropTypes.array,
    PropTypes.shape({
      name: PropTypes.string,
      total: PropTypes.string,
    }),
  ]),
  loading: PropTypes.bool
};

Accordion.defaultProps = {
  loading: false,
  children: null,
  titleData: []
};
export default Accordion;
