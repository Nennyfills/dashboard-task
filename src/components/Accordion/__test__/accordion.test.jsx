import { render, screen } from '@testing-library/react';
import Accordion from '..';

it('Should take a snapshot', () => {
  const { asFragment } = render(<Accordion children={[<div>Hello</div>]} titleData={[{ name: 'project 1', total: 576778 }]} />);
  expect(asFragment(<Accordion children={[<div>Project</div>]} titleData={[{ name: 'project1', total: 576778 }]} />)).toMatchSnapshot();
});

it('Should render a accordion with test id accordion-row', () => {
  render(<Accordion children={[<div>Hello</div>]} titleData={[{ name: 'project5', total: 576778 }]} />);
  const accordionClass = screen.getByTestId('accordion-row');
  expect(accordionClass).toBeTruthy();
  expect(accordionClass).toHaveClass('accordion-row');
});

it('Should render a navbar with test id accordion-row__col', () => {
  render(<Accordion children={[<div>Hello</div>]} titleData={[{ name: 'project2', total: 576778 }]} />);
  const accordionClass = screen.getByTestId('accordion-row__col');
  expect(screen.queryByTestId('accordion-row__col')).toBeTruthy();
  expect(accordionClass).toHaveClass('accordion-row__col');
});

it('Should render a navbar with test id accordion-row__tabs', () => {
  render(<Accordion children={[<div>Hello</div>]} titleData={[{ name: 'project3', total: 576778 }]} />);
  expect(screen.queryAllByTestId('accordion-row__tabs')).toBeTruthy();
  expect(screen.queryAllByTestId('accordion-row__tabs')).toHaveLength(2);
});

it('Should render a navbar with test id accordion-row__tab--label', () => {
  render(<Accordion children={[<div>Hello</div>]} titleData={[{ name: 'project4', total: 576778 }]} />);
  expect(screen.queryAllByTestId('accordion-row__tab--label')).toBeTruthy();
  expect(screen.queryAllByTestId('accordion-row__tab--label')).toHaveLength(1);
});

it('renders a message', () => {
  const { getByText } = render(<Accordion children={[<div>Hello</div>]} titleData={[{ name: 'project 1', total: 576778 }]} />);
  expect(getByText('Hello')).toBeInTheDocument();
  expect(getByText('project 1')).toBeInTheDocument();
  expect(getByText('576,778 USD')).toBeInTheDocument();
});
