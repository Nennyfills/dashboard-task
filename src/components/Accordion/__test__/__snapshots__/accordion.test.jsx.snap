// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should take a snapshot 1`] = `
<DocumentFragment>
  <div
    class="accordion-row"
    data-testid="accordion-row"
  >
    <div
      class="accordion-row__col"
      data-testid="accordion-row__col"
    >
      <div
        class="accordion-row__tabs"
        data-testid="accordion-row__tabs"
      >
        <div
          class="accordion-row__tab"
          data-testid="accordion-row__tabs"
        >
          <input
            class="accordion"
            id="check-0"
            type="checkbox"
          />
          <label
            class="accordion-row__tab--label semibold"
            data-testid="accordion-row__tab--label"
            date-label="amount"
            for="check-0"
          >
            <p
              class="bold"
            >
              project 1
            </p>
            <p
              class="bold"
            >
              <span>
                Total: 
              </span>
              576,778 USD
            </p>
          </label>
          <div
            class="content"
          >
            <div
              class="content--inner"
            >
              <div>
                Hello
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
